### PR TITLE
ErrBucketNotFound bugfix

### PR DESCRIPTION
**Pull request by joe94**

### CODE REVIEW

File: store/errors.go [hunk 1]
```diff
@@ -163,12 +161,8 @@ func SanitizeError(err error) error {
 		return nil
 	}
 	var errHTTP = ErrHTTPOpFailed
-
-	switch {
-	case errors.Is(err, ErrItemNotFound):
+	if errors.Is(err, ErrItemNotFound) {
```
Comment (kristinapathak) [inline store/errors.go:164]: Codecov annotated here.  I looked and it looks like we don't have any `errors_test.go` file, which means no unit tests for the code in this file? This seems tangential to this pr so I propose we make an issue and fix this later.

File: store/inmem/InMem.go [hunk 2]
```diff
@@ -37,77 +37,49 @@ func ProvideInMem() store.S {
...
-		}
+	bucket, ok := i.data[key.Bucket]
+	if !ok {
+		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "get"})
 	}
-	if err != nil {
-		err = store.ItemOperationError{Err: err, Key: key, Operation: "get"}
+	item, ok := bucket[key.ID]
+	if !ok {
+		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "get"})
```
Comment (kristinapathak) [inline store/inmem/InMem.go:57]: I understand from a program logic perspective, we don't care which case causes this error.  But is there a reason that a human might want to know when debugging? Is there a benefit of wrapping the `store.ErrItemNotFound` error with some text meant for humans to read in logs saying either the bucket or item didn't exist? ie:
```go
return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{
    Err: fmt.Errorf("%w: no bucket", store.ErrItemNotFound), 
    Key: key, Operation: "get",
})
```

File: store/errors.go (hunk 1 above)
Comment (joe94) [reply store/errors.go:164]: Sounds good 👍  Added https://github.com/xmidt-org/argus/issues/172

File: store/inmem/InMem.go (hunk 2 above)
Comment (joe94) [reply store/inmem/InMem.go:57]: I don't think so. In either case, the only action to remediate it is actually creating the item.

**Review by kristinapathak — APPROVED** [review]
:tada:

**Pull request merged**